aws/src/test/java/org/apache/iceberg/aws/TestAwsClientFactories.java

```diff
@@ -132,6 +136,41 @@ public void testLakeFormationAwsClientFactorySerializable() throws IOException {
         .isInstanceOf(LakeFormationAwsClientFactory.class);
   }
 
+  @Test
+  public void testDefaultAwsClientFactoryWithCredentialsProvider() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.AWS_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(
+        AwsProperties.S3FILEIO_CREDENTIALS_PROVIDER,
+        SystemPropertyCredentialsProvider.class.getName());
```

Comment:
Shouldn't we check if this provider is loaded? Usually for things like this I just make a fake class which throws an exception when initialized. That we we can tell the constructor was called correctly.

Reply:
@RussellSpitzer Added `Class.forName` to check class exists in the classpath. Please check. I could not find any existing API in iceberg project Spark `Utils.classIsLoadable` API

Reply:
+1 for making a fake class for this. See examples like https://github.com/apache/iceberg/blob/master/core/src/test/java/org/apache/iceberg/TestLocationProvider.java#L185

And do we need `Class.forName`? I think it will naturally throw ClassNotFound in `DynMethods`

Reply:
Jack has it right, I'm not worried this class does not exist. But I want the test to prove that it successfully loaded the class.

Using a fake class lets us know with out a doubt that the class was loaded and used as the credential provider because we can put traps in it.

Reply:
The point is not to test whether the CNFE is thrown, but to make sure the test code is dynamically loading the class specified. We wouldn't have had a way of distinguishing between SystemPropertiesCredentialProvider being loaded and any other provider being loaded. The new tests added are better in that they check (in the error message) that the class we wanted was actually loaded.

Reply:
Specifically this test just makes sure AwsClientFactories.from(properties) doesn't fail. It doesn't check that SystemProperitesCredentialsProvider is the credential provider